### ISSUE TITLE
Feature request : use Wikipedia as photo provider and get information for more places

### ISSUE THREAD

**Issue by nabilys** [organicmaps/organicmaps#3252]
It would be nice to get pictures of places we want to go thanks to Wikipedia, and get places informations for cities

Screenshot of Apple plans and organic maps for a city : 
![B8E90DD7-4BE0-4A67-8596-34DBA0168410](https://user-images.githubusercontent.com/59312735/186620460-ccea7055-b519-48b5-bd4a-7bad5472d8fe.png)
![2E70507D-3F46-4E71-B516-6B466F358295](https://user-images.githubusercontent.com/59312735/186620468-1f91d4a4-d62e-4c83-a174-cc30f49a9e10.png)

**Comment by biodranik** [comment]
This feature already kinda works if the OSM feature has a corresponding Wikipedia tag. Opening a web wiki page will show wiki pics.

**Comment by nabilys** [comment]
The link with Wikipedia only work for big cities or famous places in my phone, is there a way to help connecting a place to Wikipedia?
For the photo, it is very helpful to have them on the map, to discover new places. And if for example you’re on a trip, drawing your daily road, having to go to another website to get pictures is not very convenient.

**Comment by biodranik** [comment]
1. https://wiki.openstreetmap.org/wiki/Key:wikipedia
2. We won't store pictures in offline maps data, it will greatly increase map size.
3. If photos are loaded online, then it's more convenient to see them in the app instead of the browser, agree. But that will require a good amount of work. And photos will be available only for a limited set of features that are added to OSM and have photos. We can do it someday, but maybe there are some other, more efficient options for users.

**Comment by Jean-BaptisteC** [comment]
Yes, i confirm OM doesn't show Wikipedia articles for little cities or villages.
Examples:
https://www.openstreetmap.org/relation/86754
https://www.openstreetmap.org/relation/78828

**Comment by AvidSeeker** [comment]
I agree photos are an important feature for a map app, at least for travelers. As for places that are too insignificant to have their own Wikipedia page, they can be added to Wikidata. 

This has been already implemented in [Wikishootme](https://meta.wikimedia.org/wiki/WikiShootMe), so maybe you'll find that helpful for queries photos inside the app.

**Comment by oleg-PK** [comment]
Wikimedia.org and wikivoyage.org has photo content that can be used.

**Comment by oleg-PK** [comment]
**Problem Statement**
There are no photos in POIs, even though there are photos available under creative commons that can be displayed.  The absence of photos limits the usefulness of discovering new places or deciding which location to go to.

**Criteria**
- When first selecting a POI, in the summary view, it's visible that there are photos, but the full vertical photo is not displayed, only about the top half of the photo is displayed, to indicate that there is more info available below the photo.
- When expanding the place page from summary view to full view, the photos can be fully seen and all the other available info.
- The photos will be arranged horizontally and can be swiped left and right. 
- In the expanded view of place page, In the bottom left corner of the left most photo, display the logo for the source of photos (ex Wikimedia Commons).  The logo stays in place while the photos are swiped left and right.
- Adding photos:
      - At the end of the photos on the right is button called "Add Photo", which gives selection to take a photo with the camera or choose a photo from the person's library.
      - When a photo is selected, have a "Submit".
      - After the photo is submitted, display a message saying "Thank you for submitting the photo to Wikimedia Commons!".
      - TBD: can photos be added only to POIs which already exist in Wikimedia Commons, or can new POIs be created in Wikimedia Commons enabling adding photos to these new POIs?
- When selecting a photo, it will be displayed to fully fit the screen.
      - In full screen photo mode, have left and right arrow buttons on the left side of the screen in the center and and the right side of the screen in the center.  When pressing the buttons, they will instantly switch to the next photo, no swiping effect.
      - Can exit the full screen mode by swiping the photo down or but pressing the "Back" button which is located in the bottom left.

**Comment by biodranik** [comment]
Does it make sense to start with "Adding photos" from the beginning, to populate empty places and engage users in another community-driven project?

Technically, extracting photo links from external sources to preload them in the Place Page, or building our own photo service with some moderation and privacy in mind, are the main requirements for this issue.

**Comment by oleg-PK** [comment]
An OM photo service would give more flexibility, the challenge is content moderation, especially with photos, there is the possibility of vandalism.  Without a person dedicated to reviewing photos, there is a bit of risk.  With a 3rd party photo service, they would handle the basics.

**Comment by biodranik** [comment]
If OM becomes a target for vandalism, that would mean we are already so popular that may easily find a solution for moderation )

Can AI scanning help us here? Or community votes with good/bad ratings?

**Comment by oleg-PK** [comment]
Creating a new photo service appears to be a big effort, it would be useful to have an estimate of hours/days for each amount of work it would be - new OM photo service, just adding Wikimedia photos.

Wikipedia already has photos for many locations.  From the user perspective, in the short term people would find more value from seeing existing Wikipedia photos, as opposed to starting with zero photos and in a few years having the photos built up.

**Comment by biodranik** [comment]
Before deciding on priorities, let's investigate what are our options, and evaluate them.

- What are the available image sources? What are other apps using? Does the license allow it?
- Can we explicitly ask some services for permission to use their images for free?
- Did someone already implement extraction/integration with external image sources? Can we learn/reuse the implementation?
- Can we start with something smaller? What would be the roadmap?
- How much time will it take to implement?

- Can we add support for user-uploaded images in parallel?
- What are the main challenges there? What are their solutions?
- Are there any projects/code that we can reuse for this task? How is it implemented in other apps?
- Are there existing open services that we can use/promote? Will they survive scaling to tens/hundreds of millions of users?
- What are the pros and cons of recreating our own photo service?
- What is the approximate roadmap and estimate?

With clear answers to these questions, it will be easy to decide on how to proceed.

**Comment by oleg-PK** [comment]
This analysis is absolutely necessary.  It will take time and effort to complete the analysis, so before doing that work, it's a good idea to see where this issues falls on the priority list, even if just rough priority, to avoid doing work that will have no results.

**Comment by biodranik** [comment]
Loading photos for many places will make migration from other major apps easier. But only in case if there are enough photos to satisfy users.

Working on something that is rarely used or doesn't help the majority of users have a lower priority.

**Comment by RedAuburn** [comment]
maybe we could store 1-2 thumbhashes in the MWM for a poi, then load the images when pressed:

**Comment by grahamjpark** [comment]
Dropping some thoughts here as it's the most related issue:

I think dynamically extracting images from a restaurant's website would be super helpful. It would be quicker and more streamlined than opening the website for each place you're considering eating. Obviously it's not the same as user-submitted photos, but it still provides a lot of information about the food and atmosphere. Perhaps beneath the website there could be a "Retrieve website images" button.

It's uncertain to me if extracting and displaying images like this is ok legally, but search engines do something similar so there's got to be a way to implement it.

**Comment by Misalf-git** [comment]
> I think dynamically extracting images from a restaurant's website would be super helpful.

I guess a bogus URL or malicious website could then provide wrong/bad/evil content until after
1.) a data update in OSM and
2.) a map update with new data for Organic Maps.

**Comment by grahamjpark** [comment]
> > I think dynamically extracting images from a restaurant's website would be super helpful.
> 
> I guess a bogus URL or malicious website could then provide wrong/bad/evil content until after
> 1.) a data update in OSM and
> 2.) a map update with new data for Organic Maps.

That's very fair. Although in part the same holds true for the current approach. I visit random websites from OSM data and hope there's not malicious content.

In theory extracting images (and not running random javascript) may be safer if implemented correctly.
